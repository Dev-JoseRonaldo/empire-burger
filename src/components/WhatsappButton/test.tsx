import { render, screen } from '@testing-library/react'

import { WhatsappButton, WhatsappButtonProps } from '.'

const { getByRole, getByText } = screen

const data: WhatsappButtonProps = {
  name: 'contato',
  urlIcon: 'https://res.cloudinary.com/whatsapp.svg',
  urlRedirect: 'https://www.whatsapp.com/',
  alt: 'alt text'
}

describe('<WhatsappButton />', () => {
  it('should render by default', () => {
    render(<WhatsappButton {...data} />)

    const name = getByText('contato')
    const link = getByRole('link')
    const image = getByRole('img', { name: 'alt text' })

    expect(name).toBeInTheDocument()
    expect(link).toHaveAttribute('href', 'https://www.whatsapp.com/')
    expect(image).toHaveAttribute(
      'src',
      'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'
    )
    expect(image).toHaveAttribute('alt', 'alt text')
  })

  it('should match snapshot', () => {
    const { container } = render(<WhatsappButton {...data} />)

    expect(container).toMatchSnapshot()
  })
})
