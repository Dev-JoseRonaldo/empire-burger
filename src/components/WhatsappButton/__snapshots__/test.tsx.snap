// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhatsappButton /> should match snapshot 1`] = `
<div>
  <a
    class="flex gap-[0.625rem] items-center px-4 py-1 rounded bg-secondary w-min"
    href="https://www.whatsapp.com/"
    rel="noreferrer"
    target="_blank"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 22px; height: 22px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="alt text"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <span
      class="font-bold text-lg capitalize text-gray-700"
    >
      contato
    </span>
  </a>
</div>
`;
